### PR TITLE
Replace cui-modal with oui-modal in cloud module

### PR DESCRIPTION
**Pull request by radireddy**
### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions

## Title of the Pull Requests 
Replace cui-modal with oui-modal in cloud module

### Description of the Change
1. Replaced  cui-modal with oui-modal in cloud project management and open stack
2. Added loading to modal as per UX design and guidelines
3. Added validation as per UX design and guidelines

### Benefits
One manager

### Possible Drawbacks
none

### Applicable Issues
SCDC-824

### CODE REVIEW

File: client/app/cloud/project/billing/rights/addRights/cloud-project-billing-rights-add.html [hunk 1]
```diff
@@ -1,58 +1,33 @@
...
+            <input type="text"
+                id="billingContactInput"
+                name="billingContactInput"
+                class="oui-input"
+                data-ng-class="{'oui-input_error': form.$submitted && form.billingContactInput.$invalid}"
+                data-ng-model="ctrl.right.contact"
+                required>
+        </oui-field>
+        <oui-field data-label="{{ ::'cpb_rights_table_rights' | translate }}">
+            <label class="oui-select">
```
Comment (marie-j) [inline client/app/cloud/project/billing/rights/addRights/cloud-project-billing-rights-add.html]: You could use a `<oui-select>` instead

File: client/app/cloud/project/billing/rights/addRights/cloud-project-billing-rights-add.html [hunk 2]
```diff
@@ -1,58 +1,33 @@
...
+                <select id="selectRightType"
+                    name="selectRightType"
+                    class="oui-select__input"
+                    data-ng-model="ctrl.right.type">
+                    <option data-ng-repeat="type in ['readOnly', 'readWrite']" value="{{type}}">
+                        {{ ('cpb_rights_table_rights_value_' + type) | translate }}
```
Comment (marie-j) [inline client/app/cloud/project/billing/rights/addRights/cloud-project-billing-rights-add.html]: Missing `track by` and also prefer using `ng-bind`

File: client/app/cloud/project/openstack/users/add/openstack-users-add.controller.js [hunk 3]
```diff
@@ -1,37 +1,39 @@
...
+                this.OvhApiCloud.Project().User().v6().save({
+                    serviceName: this.serviceName
+                }, {
+                    description: this.model.value
+                }).$promise
+                    .then(newUser => {
+                        this.OpenstackUsersPassword.put(this.serviceName, newUser.id, newUser.password);
+                        this.CloudMessage.success(this.$translate.instant("openstackusers_users_userlist_add_submit_success"));
+                    })
+                    .catch(err => this.CloudMessage.error([this.$translate.instant("openstackusers_users_userlist_add_submit_error"), err.data && err.data.message || ""].join(" ")))
```
Comment (marie-j) [inline client/app/cloud/project/openstack/users/add/openstack-users-add.controller.js]: You can pass an argument in translation like `<translation> Lorem ipsum {{ myParam }}</translation>` and then directly call `this.$translate.instant(id, {myParam})`

File: client/app/cloud/project/billing/rights/addRights/cloud-project-billing-rights-add.html [hunk 4]
```diff
@@ -1,58 +1,33 @@
...
-        </button>
-        <button type="submit"
-            class="oui-button oui-button_primary"
-            data-ng-disabled="!ctrl.model.isAdmin || !ctrl.right.contact || ctrl.loader"
-            data-ng-click="ctrl.validateAddRight()"
-            data-translate="cpb_rights_add">
-        </button>
-    </cui-modal-footer>
-</cui-modal>
+<form novalidate name="form" data-ng-submit="form.$valid && ctrl.validateAddRight()">
```
Comment (marie-j) [inline client/app/cloud/project/billing/rights/addRights/cloud-project-billing-rights-add.html]: Maybe you could switch from `ctrl` to `$ctrl` like in the other files 😃

File: client/app/cloud/project/openstack/users/add/openstack-users-add.controller.js (hunk 3 above)
Comment (radireddy) [reply client/app/cloud/project/openstack/users/add/openstack-users-add.controller.js]: Your point is very valid but the scope of this work is to replace cui-modal with oui-modal. Your suggestion needs changes in all language translation files and it takes more time. Currently, we have not accounted for this in the current sprint. We can have a different jira story for this kind of changes and take it in next sprint.

File: client/app/cloud/project/billing/rights/addRights/cloud-project-billing-rights-add.html [hunk 5]
```diff
@@ -1,33 +1,32 @@
...
-                    </option>
-                </select>
-                <i class="oui-icon oui-icon-chevron-down" aria-hidden="true"></i>
-            </label>
+            <oui-select name="selectRightType" 
+                    id="selectRightType"
+                    model="$ctrl.right.type" 
+                    data-title="{{ ('cpb_rights_table_rights_value_' + type) | translate }}" 
+                    placeholder="{{ ('cpb_rights_table_rights_value_' + type) | translate }}" 
+                    items="['readOnly', 'readWrite']"
```
Comment (FredericEspiau) [inline client/app/cloud/project/billing/rights/addRights/cloud-project-billing-rights-add.html]: `data-`

File: client/app/cloud/project/billing/rights/addRights/cloud-project-billing-rights-add.html [hunk 6]
```diff
@@ -1,33 +1,32 @@
...
-            </label>
+            <oui-select name="selectRightType" 
+                    id="selectRightType"
+                    model="$ctrl.right.type" 
+                    data-title="{{ ('cpb_rights_table_rights_value_' + type) | translate }}" 
+                    placeholder="{{ ('cpb_rights_table_rights_value_' + type) | translate }}" 
+                    items="['readOnly', 'readWrite']" 
+                    required 
+                    data-align="start">
+                <span ng-bind="$item"></span>
```
Comment (FredericEspiau) [inline client/app/cloud/project/billing/rights/addRights/cloud-project-billing-rights-add.html]: `data-`

File: client/app/cloud/project/billing/rights/addRights/cloud-project-billing-rights-add.html [hunk 7]
```diff
@@ -1,58 +1,32 @@
...
+<form novalidate name="form" data-ng-submit="form.$valid && $ctrl.validateAddRight()">
+    <oui-modal 
+        data-title="{{:: 'cpb_rights_add_title' | translate}}"
+        data-primary-label="{{ ::'cpb_rights_add' | translate }}"
+        data-secondary-action="$ctrl.cancel()"
+        data-secondary-label="{{ ::'common_cancel' | translate }}"
+        data-on-dismiss="$ctrl.cancel()"
+        data-loading="$ctrl.loader">
+         
+        <oui-field data-label="{{ $ctrl.model.isUSorCA ? 'cpb_rights_table_email' : 'cpb_rights_table_nichandle' | translate }}">
```
Comment (FredericEspiau) [inline client/app/cloud/project/billing/rights/addRights/cloud-project-billing-rights-add.html]: `::`

File: client/app/cloud/project/billing/rights/addRights/cloud-project-billing-rights-add.html [hunk 8]
```diff
@@ -1,58 +1,32 @@
...
+                class="oui-input"
+                data-ng-class="{'oui-input_error': form.$submitted && form.billingContactInput.$invalid}"
+                data-ng-model="$ctrl.right.contact"
+                required>
+        </oui-field>
+        <oui-field data-label="{{ ::'cpb_rights_table_rights' | translate }}">
+            <oui-select name="selectRightType" 
+                    id="selectRightType"
+                    data-model="$ctrl.right.type" 
+                    data-title="{{ ('cpb_rights_table_rights_value_' + type) | translate }}"
```
Comment (FredericEspiau) [inline client/app/cloud/project/billing/rights/addRights/cloud-project-billing-rights-add.html]: `::`

File: client/app/cloud/project/billing/rights/addRights/cloud-project-billing-rights-add.html [hunk 9]
```diff
@@ -1,58 +1,32 @@
...
+                data-ng-class="{'oui-input_error': form.$submitted && form.billingContactInput.$invalid}"
+                data-ng-model="$ctrl.right.contact"
+                required>
+        </oui-field>
+        <oui-field data-label="{{ ::'cpb_rights_table_rights' | translate }}">
+            <oui-select name="selectRightType" 
+                    id="selectRightType"
+                    data-model="$ctrl.right.type" 
+                    data-title="{{ ('cpb_rights_table_rights_value_' + type) | translate }}" 
+                    data-placeholder="{{ ('cpb_rights_table_rights_value_' + type) | translate }}"
```
Comment (FredericEspiau) [inline client/app/cloud/project/billing/rights/addRights/cloud-project-billing-rights-add.html]: `::`

File: client/app/cloud/project/billing/vouchers/addCredit/cloud-project-billing-vouchers-add-credit.controller.js [hunk 10]
```diff
@@ -17,7 +17,7 @@ class CloudProjectBillingVouchersAddcreditCtrl {
     }
 
     addCredit () {
-        this.$uibModalInstance.close(this.credit.amount);
+        return this.$uibModalInstance.close(this.credit.amount);
```
Comment (JeremyDec) [inline client/app/cloud/project/billing/vouchers/addCredit/cloud-project-billing-vouchers-add-credit.controller.js]: You do need to return the result ? (close method returns something ?)

**Comment by antleblanc** [comment]
@radireddy Please pay attention to your commit message (ref: https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commits)

**Pull request merged**